docs/aws/audit/codebuildmonitoring/rules/codebuild_project_source_repo_url.mdx

```diff
@@ -20,10 +20,245 @@ Security
+4. In the build project details page, check the "Source" section. If the "Repository URL" field contains any sign-in credentials (username and password), then it is a misconfiguration. The URL should not contain any credentials.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can list all the CodeBuild projects in your account using the following command:
+
+   ```
+   aws codebuild list-projects --region <region-name>
+   ```
+
+   Replace `<region-name>` with the name of the AWS region where your CodeBuild projects are located.
+
+3. After getting the list of projects, you can describe each project to get its details. Use the following command to do this:
+
+   ```
+   aws codebuild batch-get-projects --names <project-name> --region <region-name>
+   ```
+
+   Replace `<project-name>` with the name of your CodeBuild project and `<region-name>` with the name of the AWS region where your CodeBuild project is located.
+
+4. In the output of the above command, look for the `source` field. This field contains the information about the source repository of the CodeBuild project. If the `type` field under `source` is `BITBUCKET` and the `location` field contains the sign-in credentials, then the CodeBuild project is misconfigured. The `location` field should not contain the sign-in credentials.
+</Accordion>
 
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, you need to install the AWS SDK for Python (Boto3) and GitPython. You can install these libraries using pip:
+
+   ```python
+   pip install boto3
+   pip install GitPython
+   ```
+
+2. Connect to AWS CodeBuild: Use the Boto3 library to connect to AWS CodeBuild. You will need your AWS access key ID, secret access key, and the region where your CodeBuild project is located.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+
+   codebuild = session.client('codebuild')
+   ```
+
+3. Retrieve and analyze the CodeBuild project configuration: Use the `batch_get_projects` method to retrieve the configuration of your CodeBuild project. Then, check if the source repository URL contains any sign-in credentials.
+
+   ```python
+   response = codebuild.batch_get_projects(names=['YOUR_PROJECT_NAME'])
+
+   for project in response['projects']:
+       source = project['source']
+       if 'auth' in source:
+           auth = source['auth']
+           if auth['type'] == 'OAUTH' and 'resource' in auth:
+               print(f"Sign-in credentials found in source repository URL: {auth['resource']}")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>